### ISSUE TITLE
Add support for Vault KV-v2 api

### ISSUE THREAD

**Issue by kadary** [vert-x3/vertx-config#85]
Actually only Vault KV-v1 (https://www.vaultproject.io/api/secret/kv/kv-v1.html) is supported for vault config store. 

It will be great to add support for vault KV-v2 API (https://www.vaultproject.io/api/secret/kv/kv-v2.html).

The main difference between the two versions is data structure. 
Eg: when reading secrets:
V1:
```
{
  "auth": null,
  "data": {
    "foo": "bar",
    "ttl": "1h"
  },
  "lease_duration": 3600,
  "lease_id": "",
  "renewable": false
}
```
V2:
```
{
  "data": {
    "data": {
        "foo": "bar"
    },
    "metadata": {
      "created_time": "2018-03-22T02:24:06.945319214Z",
      "deletion_time": "",
      "destroyed": false,
      "version": 1
    }
  },
}
```

I can do a PR for this feature if you think it can be integrated in vault config store.

**Comment by kadary** [comment]
Normally Fix by PR #86

Waiting for validation

**Comment by kadary** [comment]
PR has been merged.
